app/src/main/java/com/example/poste/activities/DashboardActivity.java

```diff
@@ -98,37 +101,11 @@ private void createUpdateCallback() {
              */
             @Override
             public void onSuccess() {
-                userFolders = currentUser.getFolders(); // update our local copy of the folders
-                // Fill folder view (Recycler View)
-                // Note: This is not the best way to do this, because it means every time
-                // DashboardActivity restarts, a new Adapter is created, but it works for now. The
-                // better way would be to use DiffUtil to calculate the difference between the old
-                // and new lists, and then update the RecyclerView accordingly using
-                // AdapterListUpdateCallback. This method is an attempt to keep the code simple
-                // For more information, see:
-                // https://developer.android.com/reference/androidx/recyclerview/widget/DiffUtil
-                // and
-                // https://developer.android.com/reference/androidx/recyclerview/widget/ListUpdateCallback
-                folderAdapter = new FolderAdapter(
-                        new FolderAdapter.ClickListener() {
-                            @Override
-                            public void onItemClick(int position, Folder model) {
-                                // Send to that folder's view
-                                User.getUser().setSelectedFolder(model);
-                                Intent intent = new Intent(DashboardActivity.this, FolderViewActivity.class);
-                                startActivity(intent);
-                            }
-
-                            @Override
-                            public void onItemLongClick(int position, Folder model) {
-                                // Prompt to share folder
-                                Toast.makeText(DashboardActivity.this, getString(R.string.share_folder), Toast.LENGTH_LONG).show();
-                            }
-                        },
-                        new ArrayList<>(userFolders)
-                );
-                folderRecyclerView.setAdapter(folderAdapter);
-                registerForContextMenu(folderRecyclerView);
+                folderAdapter.setLocalDataSet(User.getUser().getFolders());
+                folderAdapter.notifyDataSetChanged();
+                if (User.getUser().getFolders().size() == 0) {
+                    dashboardViewEmptyText.setVisibility(View.VISIBLE);
+                }
```

Comment:
This enables the empty dash notification, if necessary. Avoids popping in and out.

Reply:
Will it work on a failed to update also?

Reply:
Yes, as it is also set in the onError method of the callback